### PR TITLE
added basket/sqs newsletter signup + signup route + tests

### PR DESCRIPTION
**Pull request by Pomax**
closes https://github.com/mozilla/network-pulse-api/issues/496 by adding the same basket/sqs signup process we use on the foundation site, with a `/signup` route and test to verify the route works (both when called correctly, and when called in several incorrect ways)

### CODE REVIEW

File: pulseapi/settings.py [hunk 1]
```diff
@@ -31,6 +31,8 @@
     CORS_ORIGIN_REGEX_WHITELIST=(list, []),
     CORS_ORIGIN_WHITELIST=(list, []),
     CSRF_TRUSTED_ORIGINS=(list, []),
+    CRM_AWS_SQS_ACCESS_KEY_ID=(str, None),
```
Comment (Pomax) [inline pulseapi/settings.py]: Do these need to go in the `sample.env`?

File: pulseapi/urls.py [hunk 2]
```diff
@@ -65,6 +68,9 @@
 
     # Autocomplete
     url(r'^ajax_select/', include(ajax_select_urls)),
+
+    # Newsletter signup
+    url(r'signup/', signup_submission_view, name='newsletter-signup'),
```
Comment (Pomax) [inline pulseapi/urls.py]: This did not seem appropriate to wrap in the api versioning system.

File: pulseapi/settings.py [hunk 3]
```diff
@@ -346,6 +348,9 @@ def show_toolbar(request):
     MEDIA_ROOT = root('media/')
     MEDIA_URL = '/media/'
 
+CRM_AWS_SQS_ACCESS_KEY_ID = env('CRM_AWS_SQS_ACCESS_KEY_ID')
+CRM_PETITION_SQS_QUEUE_URL = env('CRM_PETITION_SQS_QUEUE_URL')
```
Comment (cadecairos) [inline pulseapi/settings.py]: we can probably rename these: `AWS_SQS_ACCESS_KEY_ID` and `SQS_QUEUE_URL`, We also need an `AWS_SQS_SECRET_ACCESS_KEY`

File: pulseapi/utility/newsletters/signup.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    'client': SQSProxy()
+}
+
+# But if there's an SQS access key, bind a "real client".
+if settings.CRM_AWS_SQS_ACCESS_KEY_ID:
+    crm_sqs['client'] = boto3.client(
+        'sqs',
+        region_name=settings.CRM_AWS_SQS_REGION,
+        aws_access_key_id=settings.CRM_AWS_SQS_ACCESS_KEY_ID,
+        aws_secret_access_key=settings.CRM_AWS_SQS_SECRET_ACCESS_KEY,
```
Comment (cadecairos) [inline pulseapi/utility/newsletters/signup.py]: These variable names will need updating too.

File: pulseapi/settings.py (hunk 1 above)
Comment (cadecairos) [reply pulseapi/settings.py]: shouldn't be necessary, since environ is setting None when they're not defined in the environment.

Comment (cadecairos) [reply pulseapi/settings.py]: on second thought, might be helpful to have them there, as empty strings with a quick description or something?

Comment (Pomax) [reply pulseapi/settings.py]: added

**Comment by gideonthomas** [comment]
@Pomax are we ready to land this? Can you add/link to documentation on how to use this route?

cc @mmmavis this is the route you will use for #506

**Comment by Pomax** [comment]
Yeah, good to land. @cadecairos could I ask you to add the new vars to the staging/prod apps?

**Comment by alanmoo** [comment]
@Pomax No, you can't. He's on leave for 3 months 😛. Do you have the permissions you need to do it yourself?

File: sample.env [hunk 5]
```diff
@@ -1,8 +1,16 @@
 DEBUG=True
-LOGIN_ALLOWED_REDIRECT_DOMAINS=test.example.com:3000
```
Comment (mmmavis) [inline sample.env]: change history of this file: https://github.com/mozilla/network-pulse-api/commits/master/sample.env

**Pull request merged**